### PR TITLE
npm-publisher: Use projects babel config

### LINKED ISSUE #196: monorepo-npm-publisher: support custom babel.config.js
I would like to use `@adeira/monorepo-npm-publisher` on different project, but it needs to be transpiled with additional babel plugins

Proposed solution:
- publisher will try to find `babel.config.js`
- if found, it will automatically add `'@adeira/babel-preset-adeira` if it's not present in presets
- if config already contains `'@adeira/babel-preset-adeira` preset, it will be replaced as follow to enable multiple targets: 
```
[
          '@adeira/babel-preset-adeira',
          {
            target,
          },
 ],
```

### PR DESCRIPTION
**Pull request by tbergq**
Use rootMode upwards to apply the babel config from the project
using this package, rather than force the @adeira/babel-preset

closes #196

### CODE REVIEW

**Comment by tbergq** [comment]
I tested it by creating a new monorepo, where I added `@kiwicom/orbit-components` with `@kiwicom/babel-plugin-orbit-components` and the transpiled version build `.mjs`, `.js` and `.js.flow` and also applied the changes from `@kiwicom/babel-plugin-orbit-components`. `import { Button } from '@kiwicom/orbit-components'` was transpiled to `var _Button = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Button"));`

File: src/monorepo-npm-publisher/src/transformFileVariants.js [hunk 1]
```diff
@@ -11,9 +12,9 @@ export default function transformFileVariants(
   transpileESM: boolean,
 ): void {
   const getBabelConfig = (target: 'js' | 'js-esm' | 'flow') => {
-    // TODO: load Babel config from the packages instead (if exists)
     return {
-      root: __dirname, // do not lookup any other Babel config
+      root: findMonorepoRoot(),
+      rootMode: 'upward',
```
Comment (jaroslav-kubicek) [inline src/monorepo-npm-publisher/src/transformFileVariants.js]: if you set root as monorepo root, isn't `rootMode=upward` same as `rootMode=root`?

Comment (tbergq) [reply src/monorepo-npm-publisher/src/transformFileVariants.js]: Yes, I think you are right, I will remove the `root`

**Comment by jaroslav-kubicek** [comment]
I'm just curious about last thing - how it works when adeira preset is already in the config?

**Comment by tbergq** [comment]
I ran it in our monorepo with dryRun: true. And as far as I could tell, it still worked as expected.

**Pull request merged**